std/algorithm/comparison.d

```diff
@@ -1568,6 +1568,15 @@ if (T.length >= 2)
     return cast(typeof(return)) (chooseB ? b : a);
 }
 
+/*****
+ * Handle the common case without all the template expansions
+ * of the general case
+ */
+T max(T, U)(T a, U b) if (is(T == U) && is(typeof(a < b)))
```

Comment:
If `is(T == U)` why not simply:

```suggestion
T max(T)(T a, T b) if (is(typeof(a < b)))
```

?

Reply:
I tried that. It doesn't work:
```
T max(T)(T a, T b) { return 0; }

bool max(T...)(T t) { return true; }

void test()
{
    int a = -10;
    uint f = 10;
    pragma(msg, typeof(max(f, a)));
}
```
prints `uint`